modules/file_manager.js

```diff
@@ -64,6 +64,66 @@ GetPath(){
 
     let fullpath = path + "/";
     return fullpath;
+},
+
+/**
+ * moves a file to a new location
+ * returns if oldPath is not a file
+ * creates directories if needed
+ * @param {string} oldPath origin path
+ * @param {string} newPath destination path
+ */
+async Move(oldPath, newPath) {
+    if (!await FileExists(oldPath))
+        return;
+
+    let newDirectory = global.path.dirname(newPath);
+    if (!await PathExists(newDirectory))
+        await fsPromises.mkdir(newDirectory, { recursive: true });
+
+    await fsPromises.rename(oldPath, newPath);
+}
+
+};
+
+/**
+ * checks if a file exists
+ * (and is a file)
+ * @param {string} path path to check
+ * @returns {boolean} true if file exists (and is a file), otherwise false
+ */
+async function FileExists(path) {
+    return await Exists(async () => {
+        const stats = await fsPromises.stat(path);
+        return stats && stats.isFile();
+    });
+}
+
+/**
+ * checks if a path exists
+ * @param {string} path path to check
+ * @returns {boolean} true if path exists, otherwise false
+ */
+async function PathExists(path) {
+    return await Exists(async () => { await fsPromises.stat(path); return true; });
+}
+/**
+ * catches NotFound-errors 
+ * 
+ * used to check files/pathes and handle NotFound-errors as 'does not exist'
+ * @param {BoolAction} func function to execute
+ * @returns {boolean} false if NotFound-error occoured, otherwise result from func will be returned
+ */
+async function Exists(func) {
+    try {
+        return await func();
+    } catch (e) {
+        if (!IsNotFoundError(e))
+            throw e;
+        return false;
+    }
 }
 
-};
\ No newline at end of file
+function IsNotFoundError(e) {
+    return e.code === "ENOENT";
+}
```

Comment:
Oh file_manager isn't meant to be for files in general but for the files of mods? 
Maybe mod_file_manager would be a better name? or just mod_files? Since I like to avoid words like "manager" (they dont really add information to the class; you can add "Manager" to most class/module-name without changing it's meaning, they all are managing something ^^)
But I'm getting side tracked ^^ you have a good idea for the new modulename? 
Maybe these new functions shouldn't  even be in a new module, but extend the build in fs module? so you can do the build in global.fsPromises.mkdir(...), global.fsPromises.rename(...), etc. as well as the new global.fsPromises.move(...), global.fsPromises.fileExists(...), etc.?
Then "fs_extensions" could be a good name?

Reply:
Yea i think an extension to fs would be good. Do also try to share requires when you can like i am via global to prevent wasteful loading.